Core/src/main/java/org/gusdb/fgputil/collection/FixedSizeStringMap.java

```diff
@@ -0,0 +1,144 @@
+package org.gusdb.fgputil.collection;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.gusdb.fgputil.Tuples.TwoTuple;
+
+public class FixedSizeStringMap implements Map<String,String> {
+
+  public static class Builder {
+
+    private final Map<String,Integer> _keyIndex;
+
+    public Builder(String[] keys) {
+      if (keys == null || keys.length == 0)
+        throw new IllegalArgumentException("FixedSizeStringMap must have at least one key.");
+      _keyIndex = new LinkedHashMap<>();
```

Comment:
You can pass the size into the map here so it doesn't have to grow when you're inserting keys